### ISSUE TITLE
Custom Classification Head

### ISSUE THREAD

**Issue by pauli31** [huggingface/peft#465]
Hi, 
is it possible to use PEFT with my custom classification head? The thing is that I have my custom classification Head and I want to fine-tune the underlying Transformer model along with the classification head.
So I would like to do something like

```
peft_config = LoraConfig(
    task_type=**????**, inference_mode=False, r=8, lora_alpha=32, lora_dropout=0.1
)

model = AutoModel.from_pretrained(model_path)
model = get_peft_model(model, peft_config)
model = MyCustomClassificationHead(model, num_classes)
```

instead of 

```
peft_config = LoraConfig(
    task_type=TaskType.SEQ_CLS, inference_mode=False, r=8, lora_alpha=32, lora_dropout=0.1
)

model = AutoModelForSequenceClassification.from_pretrained(model_path)
model = get_peft_model(model, peft_config)
```
I tried to google it but without luck.
Thank you!

**Comment by oriCodesPorter** [comment]
the same question

**Comment by adibMosharrof** [comment]
I have done something similar, where I added tokens to the vocabulary and had to finetune the embedding layer.
Please see https://github.com/huggingface/peft/issues/334#issuecomment-1561727994 for more details.

**Comment by smanikumar** [comment]
peft_config = LoraConfig(   task_type=**????**, inference_mode=False, r=8, lora_alpha=32, lora_dropout=0.1)
        model = AutoModel.from_pretrained(model_path)
        model = get_peft_model(model, peft_config)
        optimizer = torch.optim.AdamW([*model.parameters(),*classification_head.parameters()], lr=5e-4)   
        
        outputs = model(input_ids=input_ids,attention_mask=attention_mask,output_hidden_states=True)
        sequence_lengths = (torch.eq(input_ids, pad_id).int().argmax(-1) - 1)      
        pooled_logits = outputs['hidden_states'][-1][torch.arange(input_ids.shape[0]), sequence_lengths]
        logits = classification_head(pooled_logits)

        loss_fct = MSELoss()  
        loss = loss_fct(logits.view(-1), labels.view(-1))

        loss.backward()
        optimizer.step()
        optimizer.zero_grad()
